### DISCUSSION TITLE
Debugging on K8s cluster

### DISCUSSION THREAD

**Discussion by slamer59** [dagster-io/dagster#7705] [Q&A] [answered]
Debug on K8S directly
Hello, I wonder how we debug easily jobs on K8s. 

For what I understood, a gRPC server is running as a Pod when we build a Job  with the same Docker image. So if we change something in the Pod we cannot see the change on the Job unless there is a shared PV or something similar ? 
If so how would you debug faster a pipeline on K8S ? 

Thanks to a technology like Remote Container, we can develop directly on the Pod. But not sure the change are made. 

How would you do ?

**Comment by johannkm** [comment, marked as answer]
Ideally you can leverage Dagster so that you can develop jobs locally, then deploy to k8s without a dev loop there. This usually means running Dagit locally pointed at the same file/module as what gets built in to the image that you deploy to K8s.

**Reply by slamer59** [reply]
Yes. The fact is that when it is deployed de have access to more ressources in prod (like S3 etc...). So debugging on k8s will help to not install on local cluster.

Thanks anyway
